patches/server/0924-Rewrite-chunk-system.patch

```diff
@@ -0,0 +1,19703 @@
+ 
+     public void propagateChanges() {
+-        if (this.lightQueue.isEmpty()) {
+-            return;
+-        }
+-
+-        final SkyStarLightEngine skyEngine = this.getSkyLightEngine();
+-        final BlockStarLightEngine blockEngine = this.getBlockLightEngine();
+-
+-        try {
+-            LightQueue.ChunkTasks task;
+-            while ((task = this.lightQueue.removeFirstTask()) != null) {
+-                if (task.lightTasks != null) {
+-                    for (final Runnable run : task.lightTasks) {
+-                        run.run();
+-                    }
+-                }
+-
+-                final long coordinate = task.chunkCoordinate;
+-                final int chunkX = CoordinateUtils.getChunkX(coordinate);
+-                final int chunkZ = CoordinateUtils.getChunkZ(coordinate);
+-
+-                final Set<BlockPos> positions = task.changedPositions;
+-                final Boolean[] sectionChanges = task.changedSectionSet;
+-
+-                if (skyEngine != null && (!positions.isEmpty() || sectionChanges != null)) {
+-                    skyEngine.blocksChangedInChunk(this.lightAccess, chunkX, chunkZ, positions, sectionChanges);
+-                }
+-                if (blockEngine != null && (!positions.isEmpty() || sectionChanges != null)) {
+-                    blockEngine.blocksChangedInChunk(this.lightAccess, chunkX, chunkZ, positions, sectionChanges);
+-                }
+-
+-                if (skyEngine != null && task.queuedEdgeChecksSky != null) {
+-                    skyEngine.checkChunkEdges(this.lightAccess, chunkX, chunkZ, task.queuedEdgeChecksSky);
+-                }
+-                if (blockEngine != null && task.queuedEdgeChecksBlock != null) {
+-                    blockEngine.checkChunkEdges(this.lightAccess, chunkX, chunkZ, task.queuedEdgeChecksBlock);
+-                }
+-
+-                task.onComplete.complete(null);
+-            }
+-        } finally {
+-            this.releaseSkyLightEngine(skyEngine);
+-            this.releaseBlockLightEngine(blockEngine);
+-        }
++        throw new UnsupportedOperationException("No longer implemented, task draining is now performed by the light thread"); // Paper - replace light queue
+     }
+ 
+     protected static final class LightQueue {
+diff --git a/src/main/java/com/destroystokyo/paper/io/PaperFileIOThread.java b/src/main/java/com/destroystokyo/paper/io/PaperFileIOThread.java
+index 7c89a96d54641904e2d4562fe28c59deecfb5444..5e5ecd3aac657bfd9beb3a02fa274de6fdcad707 100644
+--- a/src/main/java/com/destroystokyo/paper/io/PaperFileIOThread.java
++++ b/src/main/java/com/destroystokyo/paper/io/PaperFileIOThread.java
+@@ -42,10 +42,10 @@ public final class PaperFileIOThread extends QueueExecutorThread {
+ 
+     public static final class Holder {
+ 
+-        public static final PaperFileIOThread INSTANCE = new PaperFileIOThread();
++        public static final PaperFileIOThread INSTANCE = null; // Paper - fail hard on usage
```

Comment:
so this is why dynmap fails to work

Is this temporary, or plugins will be required to use an alternative?

Reply:
I can add back in code to keep the api hooks here for 1.19, but I wont keep them for the next release (1.20)

Reply:
If it's breaking changes then better not, i'll be able to change the dynmap to test the speed, since before it wasn't able to reach full potential of my cpu on low quality render because was waiting for raw nbt chunk data from the io thread